java-checks-test-sources/spring-3.2/src/main/java/checks/SQLInjectionCheckSample.java

```diff
@@ -1,60 +1,69 @@
+    namedParameterJdbcTemplate.query("SELECT " + input, new BeanPropertySqlParameterSource(new Object()), new RowMapperResultSetExtractor(rowMapper)); // Noncompliant
+    namedParameterJdbcTemplate.queryForList("SELECT " + input, EmptySqlParameterSource.INSTANCE); // Noncompliant
+    namedParameterJdbcTemplate.queryForMap("SELECT " + input, EmptySqlParameterSource.INSTANCE); // Noncompliant
+    namedParameterJdbcTemplate.queryForObject("SELECT " + input, EmptySqlParameterSource.INSTANCE, rowMapper); // Noncompliant
+    namedParameterJdbcTemplate.queryForRowSet("SELECT " + input, EmptySqlParameterSource.INSTANCE); // Noncompliant
+    namedParameterJdbcTemplate.queryForStream("SELECT " + input, EmptySqlParameterSource.INSTANCE, rowMapper); // Noncompliant
+    namedParameterJdbcTemplate.update("SELECT " + input, EmptySqlParameterSource.INSTANCE); // Noncompliant
+
 
     JdbcDaoImpl jdbcDao = new JdbcDaoImpl();
     jdbcDao.setAuthoritiesByUsernameQuery("SELECT " + input); // Noncompliant
+    jdbcDao.setGroupAuthoritiesByUsernameQuery("SELECT " + input); // Noncompliant
+    jdbcDao.setUsersByUsernameQuery("SELECT " + input); // Noncompliant
 
     JdbcUserDetailsManager jdbcUserDetailsManager = new JdbcUserDetailsManager();
+    jdbcUserDetailsManager.setChangePasswordSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setCreateAuthoritySql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setCreateUserSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setDeleteGroupAuthoritiesSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setDeleteGroupAuthoritySql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setDeleteGroupMemberSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setDeleteGroupMembersSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setDeleteGroupSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setDeleteUserAuthoritiesSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setDeleteUserSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setFindAllGroupsSql("SELECT " + input); // Noncompliant
     jdbcUserDetailsManager.setFindGroupIdSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setFindUsersInGroupSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setGroupAuthoritiesSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setInsertGroupAuthoritySql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setInsertGroupMemberSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setInsertGroupSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setRenameGroupSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setUpdateUserSql("SELECT " + input); // Noncompliant
+    jdbcUserDetailsManager.setUserExistsSql("SELECT " + input); // Noncompliant
```

Comment:
Two suggestions for improving the test:

1. `"SELECT " + input` appears ca. 30 times here. You could assign a variable `var sqlStatement = "SELECT " + input`, or even have `String sqlStatement` as a method parameter again, since the concrete contents is irrelevant.
2. Since the method is now quite long and has many parameters, we could split it by owners, which would a) reduce the length of the method, b) reduce the parameter list, c) give the method a more speaking name. Such as in: `public void jdbcClientMethods(String sqlStatements, JdbcClient jdbcClient) { jdbcClient.... }`

Reply:
1. Not really. That is what the rule is really about.
2. Will do! Nice suggestion.